files/preStudies/frameworks.tex

```diff
@@ -0,0 +1,80 @@
+	\end{itemize}
+	
+\textbf{Nice to have:}
+	
+	\begin{itemize}
+	\item 1 platform both for web and apps.
+	\item Type safety.
+	\item Reactive data. Updated data in the database should reflect on the client.
+	\item Compiles to native code for increased performance.
+	\end{itemize}
+	
+
+\subsubsection{Meteor}
+
+\textbf{Language:} JavaScript
+\\
+\textbf{Description:}
+\\
+\textit{Meteor} is a platform and framework for developing both web applications as well as mobile applications. It comes with \textit{Cordova} built-in and is considered super fast for developing applications and prototyping. It comes with a lot of build-in tools and a package library for easy setup of things like account/user integration and login integrations of different kinds like Google and Facebook. It also comes with hot code push which is a tool for uploading new versions of your apps to the stores. \textit{Meteor} also have its own Command Line Interface which is very useful when developing. \textit{Meteor} also allowes you to write your backend in plain javaScript. Out of the box it integrates with Mongo, a document database for data storage. It also uses DDP (Distributed Data Protocol) for sending data between the client and the server. This data is reactive and will be updated at the clients when it changes.
+\\
+\textit{Meteor} comes with its own view layer named Blaze. This can be substituted with React or AngularJS if either of them are preferred.
+
+\subsubsection{NativeScript}
+\textbf{Language:} JavaScript
+\\
+\textbf{Description:}
+\\
+\textit{NativeScript} uses a javascript toolchain to compile the javascript code into native API’s. This is a huge advantage because it makes the app work exactly like it was written natively without ever writing any native code or knowing about the native API’s. The markup is written in XML which is then styled with CSS.
+It’s developed by Telerik, a big software company where you can opt-in for support through a paid license. It’s supported by Google and therefore works well with \textit{Angular} and TypeScript. With TypeScript, the ''nice to have'' feature, type safety is fulfilled. \textit{Bluetooth Low Energy} capabilities can be achieved with the npm package \textit{nativescript-bluetooth}, created by npm user eddyverbruggen. \textit{NativeScript} supports iOS, Android, \textit{Universal Windows Platform} and if \textit{Angular} is used it also supports building web applications \citep{preStudy:frameworks:nativescript}.
+
+
+\subsubsection{ReactNative}
+\textbf{Language:} JavaScript, Objective-C and Java
+\\
+\textbf{Description:}
+\\
+\textit{ReactNative} is a framework by Facebook Inc. It extends the React framework for building native apps. \textit{ReactNative} requires the developers to know Objective-C for iOS development and Java for Android development. This does not fulfill our requirement that it has to be cross-platform without the developers needing to know about the distinct platform languages and API’s.
```

Comment:
Comma just after Facebook -> "Facebook, Inc."
React, iOS and Android in italics.
"our requirements" -> "the requirements" or "the projects requirements"
"that is has" -> "regarding"
API's -> APIs